packages/atclient/src/atkeys_file.c

```diff
@@ -145,12 +145,14 @@ int atclient_atkeys_file_from_string(atclient_atkeys_file *atkeys_file, const ch
   }
 
   cJSON *apkam_symmetric_key = cJSON_GetObjectItem(root, ATCLIENT_ATKEYS_FILE_APKAM_SYMMETRIC_KEY_JSON_KEY);
-  if (apkam_symmetric_key != NULL) {
-    if ((ret = set_apkam_symmetric_key_str(atkeys_file, apkam_symmetric_key->valuestring,
-                                          strlen(apkam_symmetric_key->valuestring))) != 0) {
-      atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "set_apkam_symmetric_key_str: %d\n", ret);
-      goto exit;
-    }
+  if (apkam_symmetric_key == NULL) {
+    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "Failed to read apkamSymmetricKey from JSON\n");
+    goto exit;
+  }
+  if ((ret = set_apkam_symmetric_key_str(atkeys_file, apkam_symmetric_key->valuestring,
+                                         strlen(apkam_symmetric_key->valuestring))) != 0) {
+    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "set_apkam_symmetric_key_str: %d\n", ret);
+    goto exit;
```

Comment:
The change is incorrect, not every keyfile will have apkamSymmetricKey set, but they are still valid files if they do not have the value.